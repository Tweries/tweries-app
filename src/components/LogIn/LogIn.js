import { faInfo } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import React from 'react';

const copy = {
  'Log in': 'Log in',
  'Login is necessary in order for your series of Tweets to be sent through your Twitter account':
    'Login is necessary in order for your series of Tweets to be sent through your Twitter account'
};

function LogIn({ loginWithRedirect }) {
  return (
    <>
      <p className="flex justify-center my-4">
        <button
          className="font-bold px-6 py-2 rounded tweries-background-color-blue-button"
          data-testid="login"
          onClick={() => loginWithRedirect({})}
          type="button"
        >
          {copy['Log in']}
        </button>
      </p>
      <p className="flex justify-center my-4">
        <FontAwesomeIcon
          className="tooltip tweries-color-dark-blue"
          icon={faInfo}
          size="1x"
        />
        <span className="p-3 mt-6 -ml-1 tooltip-text tweries-border">
          {
            copy[
              'Login is necessary in order for your series of Tweets to be sent through your Twitter account'
            ]
          }
        </span>
      </p>
    </>
  );
}

LogIn.propTypes = {
  loginWithRedirect: PropTypes.func.isRequired
};

export default LogIn;
