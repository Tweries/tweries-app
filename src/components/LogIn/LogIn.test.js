import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import LogIn from './LogIn';

describe('<LogIn />', () => {
  it('to match snapshot', () => {
    const { container } = render(<LogIn />);
    expect(container).toMatchSnapshot();
  });

  it('onClick', () => {
    const loginWithRedirect = jest.fn();

    const { getByTestId } = render(
      <LogIn loginWithRedirect={loginWithRedirect} />
    );

    fireEvent.click(getByTestId('login'));

    expect(loginWithRedirect).toBeCalled();
  });
});
