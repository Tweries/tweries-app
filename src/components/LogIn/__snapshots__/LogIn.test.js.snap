// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> to match snapshot 1`] = `
<div>
  <p
    class="flex justify-center my-4"
  >
    <button
      class="font-bold px-6 py-2 rounded tweries-background-color-blue-button"
      data-testid="login"
      type="button"
    >
      Log in
    </button>
  </p>
  <p
    class="flex justify-center my-4"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-info fa-w-6 fa-1x tooltip tweries-color-dark-blue"
      data-icon="info"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"
        fill="currentColor"
      />
    </svg>
    <span
      class="p-3 mt-6 -ml-1 tooltip-text tweries-border"
    >
      Login is necessary in order for your series of Tweets to be sent through your Twitter account
    </span>
  </p>
</div>
`;
